projects/RabbitMQ.Client/client/impl/ModelBase.cs

```diff
@@ -69,9 +69,7 @@ internal abstract class ModelBase : IModel, IRecoverable
         protected ModelBase(ConnectionConfig config, ISession session)
         {
             ContinuationTimeout = config.ContinuationTimeout;
-            ConsumerDispatcher = config.DispatchConsumersAsync ?
-                (IConsumerDispatcher)new AsyncConsumerDispatcher(this, config.DispatchConsumerConcurrency) :
-                new ConsumerDispatcher(this, config.DispatchConsumerConcurrency);
+            ConsumerDispatcher = new ConsumerDispatcher(this, config.DispatchConsumerConcurrency);
```

Comment:
This, although unlikely that it matters, will prevent users to sync dispatch anything to a type that implements IAsyncConsumer as well.